core/src/main/java/com/linecorp/armeria/client/DefaultHttpClient.java

```diff
@@ -42,22 +46,38 @@ public HttpResponse execute(HttpRequest req) {
     }
 
     private HttpResponse execute(@Nullable EventLoop eventLoop, HttpRequest req) {
-        final String originalPath = req.path();
-        final String newPath = concatPaths(uri().getRawPath(), originalPath);
-        final PathAndQuery pathAndQuery = PathAndQuery.parse(newPath);
-        if (pathAndQuery == null) {
-            req.abort();
-            return HttpResponse.ofFailure(new IllegalArgumentException("invalid path: " + newPath));
-        }
+        final URI uri = URI.create(req.path());
```

Comment:
- Could catch an exception and assume `authority == null`.
- Could call `URI.create()` only when `req.path()` contains `://`.

Reply:
Added [`isAbsolutePath()`](https://github.com/line/armeria/pull/1961/files#diff-6d9fb3249b1a725b4c80301a1f7d4373R340) method to check whether path is an absolute containing `://`.